qiskit_experiments/library/characterization/t2ramsey.py

```diff
@@ -138,26 +116,29 @@ def circuits(self) -> List[QuantumCircuit]:
 
         Returns:
             The experiment circuits
-
-        Raises:
-            ValueError: When conversion factor is not set.
         """
-        if self.backend:
-            self._set_backend(self.backend)
-        prefactor = self.experiment_options.conversion_factor
-
-        if prefactor is None:
-            raise ValueError("Conversion factor is not set.")
+        if self.backend and hasattr(self.backend.configuration(), "dt"):
+            dt_unit = True
+            dt_factor = self.backend.configuration().dt
+        else:
+            dt_unit = False
 
         circuits = []
-        for delay in prefactor * np.asarray(self.experiment_options.delays, dtype=float):
-            delay = np.round(delay, decimals=10)
+        for delay in self.experiment_options.delays:
+            if dt_unit:
+                real_delay_in_sec = delay_dt * dt_factor
```

Comment:
seems like `delay_dt` is used before assignment. how is this test passed? seems like we are testing only on the backend without dt.

Reply:
Yes, this is how we test, but I always run on a device and look at the results, so I can't explain.

Reply:
Actually I can explain: I've had other, unrelated problems with T2Ramsey-on-device lately, in the main branch, I think I opened issues. So I didn't run.